partitions.go

```diff
@@ -231,6 +253,12 @@ func (p *partitions) getPeers(partition int) []string {
 
 	peers := make([]string, len(p.remote[partition]))
 	copy(peers, p.remote[partition])
+
+	oldPeers := make([]string, 1024)
+	copy(oldPeers, p.old[partition])
+	// Append old peers to peer list, in case of Zookeeper issues.
+	peers = append(peers, oldPeers...)
```

Comment:
This means the dead peers will get shuffled into the live peers by the caller of this code. I think it'd be better if FindPeers return shuffled live peers + dead peers at the end.